idaes/models/properties/general_helmholtz/helmholtz_functions.py

```diff
@@ -154,12 +154,165 @@ class AmountBasis(enum.Enum):
+        "arg_units": [dimensionless, pyo.units.kg / pyo.units.kg, pyo.units.kPa],
+        "doc": "v(comp, delta, tau)",
+    },
+    # Functions of (h, p)
+    "u_hp_func": {  # internal energy
+        "fname": "u_hp",
+        "units": pyo.units.kJ / pyo.units.kg,
+        "arg_units": [dimensionless, pyo.units.kJ / pyo.units.kg, pyo.units.kPa],
+        "doc": "u(comp, enthalpy, pressure)",
+    },
+    "s_hp_func": {  # entropy
+        "fname": "s_hp",
+        "units": pyo.units.kJ / pyo.units.kg / pyo.units.K,
+        "arg_units": [dimensionless, pyo.units.kJ / pyo.units.kg, pyo.units.kPa],
+        "doc": "s(comp, enthalpy, pressure)",
+    },
+    "g_hp_func": {  # Gibbs free energy
+        "fname": "g_hp",
+        "units": pyo.units.kJ / pyo.units.kg,
+        "arg_units": [dimensionless, pyo.units.kJ / pyo.units.kg, pyo.units.kPa],
+        "doc": "g(comp, enthalpy, pressure)",
+    },
+    "f_hp_func": {  # Helmholtz free energy
+        "fname": "f_hp",
+        "units": pyo.units.kJ / pyo.units.kg,
+        "arg_units": [dimensionless, pyo.units.kJ / pyo.units.kg, pyo.units.kPa],
+        "doc": "f(comp, enthalpy, pressure)",
+    },
+    "cv_hp_func": {  # constant volume heat capacity
+        "fname": "cv_hp",
+        "units": pyo.units.kJ / pyo.units.kg / pyo.units.K,
+        "arg_units": [dimensionless, pyo.units.kJ / pyo.units.kg, pyo.units.kPa],
+        "doc": "cv(comp, enthalpy, pressure)",
+    },
+    "cp_hp_func": {  # constant pressure heat capacity
+        "fname": "cp_hp",
+        "units": pyo.units.kJ / pyo.units.kg / pyo.units.K,
+        "arg_units": [dimensionless, pyo.units.kJ / pyo.units.kg, pyo.units.kPa],
+        "doc": "cp(comp, enthalpy, pressure)",
+    },
+    "w_hp_func": {  # speed of sound
+        "fname": "w_hp",
+        "units": pyo.units.m / pyo.units.s,
+        "arg_units": [dimensionless, pyo.units.kJ / pyo.units.kg, pyo.units.kPa],
+        "doc": "w(comp, enthalpy, pressure)",
+    },
+    "v_hp_func": {  # specific volume
+        "fname": "v_hp",
+        "units": pyo.units.m**3 / pyo.units.kg,
+        "arg_units": [dimensionless, pyo.units.kJ / pyo.units.kg, pyo.units.kPa],
+        "doc": "v(comp, enthalpy, pressure)",
+    },
+    # Functions of (s, p)
+    "u_sp_func": {  # internal energy
+        "fname": "u_sp",
+        "units": pyo.units.kJ / pyo.units.kg,
+        "arg_units": [dimensionless, pyo.units.kJ / pyo.units.kg/ pyo.units.K, pyo.units.kPa],
+        "doc": "u(comp, entropy, pressure)",
+    },
+    "h_sp_func": {  # enthalpy
+        "fname": "h_sp",
+        "units": pyo.units.kJ / pyo.units.kg,
+        "arg_units": [dimensionless, pyo.units.kJ / pyo.units.kg/ pyo.units.K, pyo.units.kPa],
+        "doc": "h(comp, entropy, pressure)",
+    },
+    "g_sp_func": {  # Gibbs free energy
+        "fname": "g_sp",
+        "units": pyo.units.kJ / pyo.units.kg,
+        "arg_units": [dimensionless, pyo.units.kJ / pyo.units.kg/ pyo.units.K, pyo.units.kPa],
```

Comment:
These units also look wrong

Reply:
I think these are right.  The args art component (dimensionless), entropy (kJ/kg/K), and pressure kPa.  The Gibbs free energy is kJ/kg.

Reply:
The doc directly below says `"g(comp, delta, tau)"`. Maybe the variables are what are wrong?

Reply:
Oh yeah the doc string is wrong.